### ISSUE TITLE
∇conv_filter seems broken 

### ISSUE THREAD

**Issue by Keno** [FluxML/NNlib.jl#69]
I was validating my XLA implementation of the NNlib functions against the provided implemenations and failed to do so for ∇conv_filter. Upon closer inspecting, ∇conv_filter seems quite broken:

```
julia> f(dy, input, kernel) = NNlib.∇conv_filter(dy, input, kernel; pad=(0,0), stride=(1,1), dilation=(1,1))
f (generic function with 2 methods)

julia> f(Float32.(reshape(a, (2,2,1,1))), reshape(Float32[1.], (1,1,1,1)), Float32.(A))
2×2×1×1 Array{Float32,4}:
[:, :, 1, 1] =
 3.68136e-32  5.44964e-32
 3.68189e-32  4.54e-33

julia> f(Float32.(reshape(a, (2,2,1,1))), reshape(Float32[1.], (1,1,1,1)), Float32.(A))
2×2×1×1 Array{Float32,4}:
[:, :, 1, 1] =
 1.55684e-41  1.55684e-41
 1.55684e-41  1.55544e-42
```

(notice the values changing so something is reading uninitialized memory).

**Comment by tejank10** [comment]
What should be the correct output in this case? Since with `kernel` size of `(2, 2, 1, 1)` and `input` size of `(1, 1, 1, 1)`, it is expected that `dy` won't be of size `(2, 2, 1, 1)`. So it should be throwing a dimension mismatch error, right?

**Comment by staticfloat** [comment]
Fixed by #94
